cli/internal/context/context_test.go

```diff
@@ -121,3 +127,33 @@ func Test_isWorkspaceReference(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildPackageGraph_DuplicateNames(t *testing.T) {
+	path := getTestDir(t, "duplicate-pkgs")
+	pkgJSON := &fs.PackageJSON{
+		Name:           "my-monorepo",
+		PackageManager: "npm@8.1.2",
+	}
```

Comment:
one thing that's weird about this setup is that this function takes both a `fs.PackageJSON` struct _and_ a path to the monorepo, and it reads the `package.json` file again to get workspaces. Because of that, the name and PackageManager property here don't really need to match.